### PR TITLE
Action Button to Delete a Role

### PR DESCRIPTION
**Pull request by fromgonie**
A button to delete the role that is on the same row the button is.

### CODE REVIEW

File: frontend/src/views/HomeView.vue [hunk 1]
```diff
@@ -1 +1,68 @@
...
+  const dialogVisible = ref(false)
+  const roleId = ref<string>('')
+  
+  const showConfirmationDialog = (id: string) => {
+    dialogVisible.value = true;
+    roleId.value = id; 
+  }
+  
+  const deleteRole = () => {
+  fetch(`/roles/${parseInt(roleId.value)}`, {
```
Comment (vascolusitan) [inline frontend/src/views/HomeView.vue:43]: Please don't call this directly, to separate the responsibilities of the files.
Take a look at how we fetch the roles in HomeView.ts through rolesStore.fetchRoles(), and try to follow the pattern.

**Review by vascolusitan — COMMENTED** [review]
Great job!!

File: frontend/src/views/HomeView.vue [hunk 2]
```diff
@@ -1 +1,68 @@
-<template><h1>Test</h1></template>
+<template>
+    <div class="mb-4">
+      <ElButton
+        key="primary"
+        type="text"
+        @click="showConfirmationDialog(roleId)"
+        style="color: #29517A;"
```
Comment (vitorlofonseca) [inline frontend/src/views/HomeView.vue:7]: Please, don't use inline styling, use classes. All the colors you need, are in `colors.scss` file

File: frontend/src/views/HomeView.vue [hunk 3]
```diff
@@ -1 +1,68 @@
...
+    <ElDialog
+      v-model="dialogVisible"
+      title="Warning"
+      width="500"
+    >
+      <span>Role will be permanently removed. Continue?</span>
+  
+      <template #footer>
+        <div class="dialog-footer">
+          <ElButton @click="dialogVisible = false">Cancel</ElButton>
```
Comment (vitorlofonseca) [inline frontend/src/views/HomeView.vue:21]: All the logic inside the component should be inside script tag (`dialogVisible = false`)

File: frontend/src/stores/roles/state.ts [hunk 4]
```diff
@@ -4,3 +4,11 @@ import { type Ref, ref } from 'vue'
 const roles: Ref<Role[]> = ref([])
 
 export { roles }
+
+export const removeRoleFromState = (roleId: number): void => { roles.value = roles.value.filter(role => role.id !== roleId); };
```
Comment (vitorlofonseca) [inline frontend/src/stores/roles/state.ts:8]: This should be in the actions file, not the state. The state should contain just the data, without functions

File: frontend/src/theme/vars/colors.scss [hunk 5]
```diff
@@ -90,4 +92,8 @@ $color-danger: #d03333;
   --el-fill-color-dark: #{$color-white};
   --el-fill-color-darker: #{$color-white};
   --el-fill-color-blank: #{$color-white};
+
+  .deletebutton{
```
Comment (vitorlofonseca) [inline frontend/src/theme/vars/colors.scss:96]: Why do you need this class? The buttons already have the style we need

File: frontend/src/views/DeleteViewTemp.vue [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    dialogVisible.value = true;
+    roleId.value = id; 
+  }
+  
+  const onClickCancel = () => {
+    dialogVisible.value=false;
+  }
+
+async function deleteRole() {
+    try {
```
Comment (vitorlofonseca) [inline frontend/src/views/DeleteViewTemp.vue:49]: You have to create a function inside actions file. This function will responsible for 

1. calling the backend
2. Removing the role from the state

Here you should just import this action, nothing more

Check in Demo how an action is called, to fetch the roles, and base your work like that

File: frontend/src/stores/roles/actions.ts [hunk 7]
```diff
@@ -23,5 +25,18 @@ function setRoles(newRoles: Role[]): void {
   roles.value = newRoles
 }
 
+let response: any = null
```
Comment (vitorlofonseca) [inline frontend/src/stores/roles/actions.ts:28]: You don't need these two variables, do you?

File: frontend/src/views/DeleteViewTemp.vue [hunk 8]
```diff
@@ -30,8 +29,7 @@
   <script setup lang="ts">
   import { ElButton, ElDialog, ElMessage } from 'element-plus';
   import { ref } from 'vue'
-  import { deleteRequest } from '@/services/http';
-  import { removeRoleFromState } from '@/stores/roles/state';
+  import { removeRole } from '@/stores/roles/actions';
```
Comment (vitorlofonseca) [inline frontend/src/views/DeleteViewTemp.vue:32]: You don't have to import the function directly, like this. Check in DemoView how the fetch action is being used

**Review by vitorlofonseca — APPROVED** [review]

**Pull request merged**